Extension/src/LanguageServer/configurations.ts

```diff
@@ -2050,28 +2063,48 @@ export class CppProperties {
                     if (curOffset >= compilerPathStart && curOffset <= compilerPathEnd) {
                         continue;
                     }
-                    let message: string;
+                    let message: string = "";
                     if (!pathExists) {
                         if (curOffset >= forcedIncludeStart && curOffset <= forcedeIncludeEnd
-                                && !path.isAbsolute(resolvedPath)) {
+                                && !path.isAbsolute(expandedPaths[0])) {
                             continue; // Skip the error, because it could be resolved recursively.
                         }
-                        message = localize('cannot.find2', "Cannot find \"{0}\".", resolvedPath);
+                        // If there are incorrect paths and the first match includes "env:", split the string and check for environment variable.
+                        if (incorrectExpandedPaths.length > 0 && configMatches[0].includes("env:")) {
+                            const splitString = configMatches[0].split("env:");
+                            if (splitString[1]) {
+                                const matchResult = splitString[1].match(/\w+/);
+                                if (matchResult) {
+                                    const envVar = matchResult[0];
+                                    message = localize('cannot.find3', "Cannot find {0} in environment variable: {1}.", incorrectExpandedPaths.map(s => `"${s}"`).join(', '), envVar);
+                                }
+                            }
+                        // If there are incorrect paths but no "env:" in the first match, generate a message indicating the paths cannot be found.
+                        } else {
+                            let badPath = "";
+                            if (incorrectExpandedPaths.length > 0) {
+                                badPath = incorrectExpandedPaths.map(s => `"${s}"`).join(', ');
+                            } else {
+                                badPath = `"${expandedPaths[0]}"`;
+                            }
+                            message = localize('cannot.find2', "Cannot find {0}", badPath);
+                        }
+                        // Increment the count of non-existent paths.
                         newSquiggleMetrics.PathNonExistent++;
                     } else {
                         // Check for file versus path mismatches.
                         if ((curOffset >= forcedIncludeStart && curOffset <= forcedeIncludeEnd) ||
                                 (curOffset >= compileCommandsStart && curOffset <= compileCommandsEnd)) {
-                            if (util.checkFileExistsSync(resolvedPath)) {
+                            if (util.checkFileExistsSync(expandedPaths[0])) {
```

Comment:
Might be good to comment that we only check the first one because multi-path is not allowed for `forcedInclude` and `compileCommands`

Reply:
You could also improve this case by checking the length of `expandedPaths` and if it's >1, have another error message.